private/bufpkg/bufconfig/write.go

```diff
@@ -273,30 +257,173 @@ func writeConfig(
 	for _, option := range options {
 		option(writeConfigOptions)
 	}
-	if !writeConfigOptions.documentationComments && writeConfigOptions.uncomment {
-		return errors.New("cannot set uncomment without documentationComments for WriteConfig")
+	if err := validateWriteConfigOptions(writeConfigOptions); err != nil {
+		return err
 	}
-	if writeConfigOptions.moduleIdentity == nil && len(writeConfigOptions.dependencyModuleReferences) > 0 {
-		return errors.New("cannot set deps without a name for WriteConfig")
+	// This is the same default as the bufconfig getters.
+	version := V1Beta1Version
+	if writeConfigOptions.version != "" {
+		if err := validateVersion(writeConfigOptions.version); err != nil {
+			return err
+		}
+		version = writeConfigOptions.version
 	}
-	externalConfigV1 := ExternalConfigV1{
-		Version: V1Version,
+	config := &Config{
+		Version:        version,
+		ModuleIdentity: writeConfigOptions.moduleIdentity,
 	}
-	externalConfigV1.Lint.Use = defaultLintIDs
-	externalConfigV1.Breaking.Use = defaultBreakingIDs
-	if writeConfigOptions.moduleIdentity != nil {
-		externalConfigV1.Name = writeConfigOptions.moduleIdentity.IdentityString()
+	var breakingConfigVersion string
+	if writeConfigOptions.breakingConfig != nil {
+		config.Breaking = writeConfigOptions.breakingConfig
+		breakingConfigVersion = version
 	}
-	for _, dependencyModuleReference := range writeConfigOptions.dependencyModuleReferences {
-		externalConfigV1.Deps = append(
-			externalConfigV1.Deps,
-			dependencyModuleReference.String(),
-		)
+	var lintConfigVersion string
+	if writeConfigOptions.lintConfig != nil {
+		config.Lint = writeConfigOptions.lintConfig
+		lintConfigVersion = version
+	}
+	if breakingConfigVersion != lintConfigVersion {
+		return fmt.Errorf("breaking config version %q does not match lint config version %q", breakingConfigVersion, lintConfigVersion)
+	}
```

Comment:
I still don't know if this is right.

We should validate that both `breakingConfigVersion` and `lintConfigVersion` are non-empty and set to the exact same value as the `version` provided by `writeConfigOptions.version`, if any.

Otherwise you'll end up in that situation where the `writeConfigOptions.version` would incorrectly override the `{Breaking,Lint}`Config version as discussed in [this](https://github.com/bufbuild/buf/pull/766#discussion_r764439767) comment.

Reply:
Ah yes... hmm... I thought I changed this -- maybe I only changed it in the `bufmodule.go`. Let me fix this, my bad!